fuel/datasets/base.py

```diff
@@ -28,10 +29,6 @@ class Dataset(object):
         'targets')`` for MNIST (regardless of which data the data stream
         actually requests). Any implementation of a dataset should set this
         attribute on the class (or at least before calling ``super``).
-    default_iteration_scheme : :class:`.IterationScheme`, optional
```

Comment:
I wanted to propose it for two reasons really: The main thing I'm trying to do here is to simplify the code. After just 3 datasets (MNIST, CIFAR-10 and binarized MNIST) it turned out they share way too much code, so I factored a bunch of things out. In the process, I realised that `default_iteration_scheme` is barely ever used, and writing `DataStream(dataset)` is actually shorter than `dataset.get_default_stream()` for the cases where there is no iteration scheme.

For the cases where there is an actual default iteration scheme, it has started to bother me that the choice is very arbitrary. Especially considering your recent ticket #19. Should these datasets return single examples by default? Batches of size 1? A minibatch of reasonable size, considering that is what we train on in most simple settings? The choice is so arbitrary I'd rather have it made explicit to the user. Having a default only makes sense if it works in 90% of the cases, but here it really doesn't.